### PR TITLE
BL-629 subject_display subfields indexed together

### PR DESCRIPTION
**Pull request by sensei100**
BL-629 Subject display fields were getting the data mixed up
- Index named fields and description fields together so that the description fields don't go to the wrong line

### CODE REVIEW

File: lib/traject/indexer_config.rb [hunk 1]
```diff
@@ -157,7 +157,10 @@
 
 # Subject fields
 to_field "subject_facet", extract_marc("600abcdefghklmnopqrstuxyz:610abcdefghklmnoprstuvxy:611acdefghjklnpqstuvxyz:630adefghklmnoprstvxyz:648axvyz:650abcdegvxyz:651aegvxyz:653a:654abcevyz:655abcvxyz:656akvxyz:657avxyz:690abcdegvxyz", separator: " — ", trim_punctuation: true)
-to_field "subject_display", extract_subject_display
+to_field "subject_display" do |record, accumulator|
```
Comment (dkinzer) [inline lib/traject/indexer_config.rb]: @sensei100 please move this block to a function inside of custom.rb and add test for the expected behavior, thanks!

**Comment by sensei100** [comment]
@dkinzer I made the changes you requested.

File: spec/lib/traject/traject_indexer_spec.rb [hunk 2]
```diff
@@ -603,14 +603,6 @@
       end
     end
 
-    context "when a record doesn't have subject topics" do
-      let(:path) { "subject_topic_missing.xml" }
-
-      it "does not map anything to the field" do
-        expect(subject.map_record(records[0])).to eq({})
```
Comment (dkinzer) [inline spec/lib/traject/traject_indexer_spec.rb]: @sensei100 please add back a test for when there are no subjects so we don't inadvertently create a bug if we refactor this method and forget to account for records that do not have any subjects.  Thanks!

**Comment by sensei100** [comment]
Per Leanne, there would never be a case where a subfield would only have v.x.y.z.  That test does break with this new method, but since it isn't necessary I removed it.

File: spec/lib/traject/traject_indexer_spec.rb (hunk 2 above)
Comment (sensei100) [reply spec/lib/traject/traject_indexer_spec.rb]: @dkinzer That test doesn't really seem to be testing for no subjects.  The sample data includes a data field with subfields v and z.

**Comment by dkinzer** [comment]
@sensei100 sorry, I'm not following. I'm not asking to add back that specific test you removed, just to add a case for if field is missing.  Such a case may arise (for example) under dev conditions when I create a dummy record and ingest it but I do not include any subjects.  Thanks!

**Comment by dkinzer** [comment]
Sweet! I'll merge as soon as PR greens.  Thanks! 🙏 
🍬  🍬 🍬

**Pull request merged**